Tests/IdentifyInterceptorTests.m

```diff
@@ -300,27 +341,38 @@ - (void)testNullValuesInIdentifySetAreIgnoredOnActiveIdentify {
     XCTAssertNil(event2);
     XCTAssertEqual(self->_dbHelper.getLastSequenceNumber, 2);
     XCTAssertEqual(self->_dbHelper.getInterceptedIdentifyCount, 2);
+    XCTAssertEqual(self->_dbHelper.getIdentifyCount, 0);
 
     // active identify
-    AMPIdentify *identify3 = [AMPIdentify.identify add:@"add-key" value:[NSNumber numberWithInt:1]];
+    AMPIdentify *identify3 = [AMPIdentify.identify add:@"add-key" value:@1];
     NSMutableDictionary *event3 = [self getIdentifyEvent:identify3];
 
     event3 = [self->_identifyInterceptor intercept:event3];
 
     XCTAssertNotNil(event3);
     XCTAssertEqual(self->_dbHelper.getLastSequenceNumber, 3);
     XCTAssertEqual(self->_dbHelper.getInterceptedIdentifyCount, 0);
+    XCTAssertEqual(self->_dbHelper.getIdentifyCount, 1);
 
-    // check merged user properties
     NSMutableDictionary *userProperties3 = [AMPEventUtils getUserProperties:event3];
-    NSArray *userPropertiesOperations3 = [userProperties3 allKeys];
     XCTAssertNotNil(userProperties3);
-    XCTAssertEqual(userPropertiesOperations3.count, 2);
-    BOOL hasAllOperations = [[NSSet setWithArray:userPropertiesOperations3] isEqualToSet:[NSSet setWithArray:@[AMP_OP_SET, AMP_OP_ADD]]];
-    XCTAssertTrue(hasAllOperations);
-    XCTAssertTrue([userProperties3[AMP_OP_SET][@"set-key"] isEqualToString:@"set-value-a"]);
-    XCTAssertTrue([userProperties3[AMP_OP_SET][@"set-key-2"] isEqualToString:@"set-value-c"]);
-    XCTAssertNil(userProperties3[AMP_OP_SET][@"set-key-3"]);
+    NSArray *userPropertiesOperations3 = [userProperties3 allKeys];
+    XCTAssertEqual(userPropertiesOperations3.count, 1);
+    BOOL hasAddOperationsOnly = [[NSSet setWithArray:userPropertiesOperations3] isEqualToSet:[NSSet setWithArray:@[AMP_OP_ADD]]];
+    XCTAssertTrue(hasAddOperationsOnly);
+    XCTAssertTrue([userProperties3[AMP_OP_ADD][@"add-key"] isEqualToNumber:@1]);
+
+    NSArray *identifies = [_dbHelper getIdentifys:-1 limit:1];
+    NSDictionary *interceptedIdentify = [identifies lastObject];
+    NSMutableDictionary *userProperties = [AMPEventUtils getUserProperties:interceptedIdentify];
+    NSArray *userPropertiesOperations = [userProperties allKeys];
+    XCTAssertNotNil(userProperties);
+    XCTAssertEqual(userPropertiesOperations.count, 1);
```

Comment:
Maybe worth to verify the add event here compared to from the return.

Reply:
Just recommend more to verify the event from here compared to verify from the function return result, not a blocker.